### PR TITLE
Free-flying aircraft that translates and rotates about the center of gravity (rotating frame + translation)

### PR DESCRIPTION
**Pull request by ArneVoss**
## Proposed Changes
*Give a brief overview of your contribution here in a few sentences.*

Currently, the dynamic mesh module allows either a translation or a rotation. The parameters involved are:
GRID_MOVEMENT= ROTATING_FRAME or TRANSLATION_RATE
MOTION_ORIGIN= x y z
ROTATION_RATE= p q r
TRANSLATION_RATE= u v w
The farfield MACH number is set to zero and MACH_MOTION is used instead.

For example, a pull-up maneuver of an aircraft could be calculated as a steady CFD simulation and by looping around a MOTION_ORIGIN far away from the aircraft as discussed here: https://www.cfd-online.com/Forums/su2/241889-rotating_frame-grid-velocity-correct-flow-physics-wrong.html. However, a combination of pull-up and roll or yaw is not possible because the roll/yaw would require a different MOTION_ORIGIN that the pull-up.

To simulate a free-flying aircraft (in a flight mechanical sense) which moves and rotates in all six degrees of freedom about the center of gravity, a combination of both translation and rotation is beneficial. In the proposed change, the ROTATING_FRAME now considers both parameters ROTATION_RATE and TRANSLATION_RATE given in the config file. Because the default translation rates are zero, there should be no effect for users who want pure rotation.

The option STEADY_TRANSLATION now appears obsolete to me because the functionality is included in the ROTATING_FRAME, but I didn't want to remove the code because someone else might use that parameter? 

## Related Work
*Resolve any issues (bug fix or feature request), note any related PRs, or mention interactions with the work of others, if any.*

This is a follow-up of a discussion here: https://www.cfd-online.com/Forums/su2/241889-rotating_frame-grid-velocity-correct-flow-physics-wrong.html

Fix #1595 (https://github.com/su2code/SU2/issues/1588)](https://github.com/su2code/SU2/pull/1595/commits/c2018c836791944383c13d32dcfec903d5657072) is required to get proper nodal forces and has already been merged into the develop branch, thank you Pedro :)

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [x] I am submitting my contribution to the develop branch.
- [x] My contribution generates no new compiler warnings (try with the '-Wall -Wextra -Wno-unused-parameter -Wno-empty-body' compiler flags, or simply --warnlevel=2 when using meson).
- [x] My contribution is commented and consistent with SU2 style.
- [x] I have added a test case that demonstrates my contribution, if necessary.
- [x] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp) , if necessary.

### CODE REVIEW

**Comment by pcarruscag** [comment]
Using just STEADY_TRANSLATION skips computing the rotating source terms, and all the GCL corrections.

**Comment by ArneVoss** [comment]
I am not sure if a 2D case is a good example / test case for the ROTATING_FRAME option, because the cross product makes only sense in the 3D case. For example, there is no pitch axis, the y-axis actually goes in z-direction and so on...

**Comment by pcarruscag** [comment]
Makes sense, but a 3D case might be too expensive to run.
The 2D would at least ensure the rotation and translation are being accounted for.

File: config_template.cfg [hunk 1]
```diff
@@ -403,6 +403,13 @@ PRANDTL_TURB= 0.90
 %
 % Type of dynamic mesh (NONE, RIGID_MOTION, ROTATING_FRAME,
 %                       STEADY_TRANSLATION, GUST)
+% ROTATING_FRAME: This option considers both parameters ROTATION_RATE and
+% TRANSLATION_RATE, which allows to simulate a free-flying aircraft (in a flight
+% mechanical sense) which moves and rotates in all six degrees of freedom about
+% the center of gravity. In this context, the farfield MACH number is set to 
+% zero and MACH_MOTION is used instead.
```
Comment (pcarruscag) [inline config_template.cfg:410]:
```suggestion
% zero and MACH_MOTION is used instead to compute force coefficients. 
```

**Comment by ArneVoss** [comment]
I see that the test pywrapper_translating_naca0012 fails, maybe because the reference results calculated on my computer deviate slightly?

File: TestCases/py_wrapper/translating_NACA0012/config.cfg [hunk 2]
```diff
@@ -1,16 +1,20 @@
 % PROBLEM DEFINITION
 %
 SOLVER= EULER
-REF_DIMENSIONALIZATION= FREESTREAM_VEL_EQ_MACH
+REF_DIMENSIONALIZATION= DIMENSIONAL
```
Comment (pcarruscag) [inline TestCases/py_wrapper/translating_NACA0012/config.cfg:4]:
```suggestion
REF_DIMENSIONALIZATION= FREESTREAM_VEL_EQ_MACH
```

**Review by pcarruscag — APPROVED** [review]
Thanks for updating the testcase. Let's keep the non dimensionalization to make sure the python forces are still being computed properly in that case.
Yep small differences between local and github are common (compiler differences, etc.) you can take the github values to update the reference file.

File: TestCases/py_wrapper/translating_NACA0012/forces_0.csv.ref [hunk 3]
```diff
@@ -1,200 +1,200 @@
...
+28, 1.01, -8.84, 0.00
+29, 1.86, -16.61, 0.00
+30, 2.72, -24.66, 0.00
+31, 3.58, -32.98, 0.00
+32, 4.43, -41.60, 0.00
+33, 5.28, -50.47, 0.00
+34, 6.11, -59.56, 0.00
+35, 6.93, -68.92, 0.00
+36, 7.73, -78.52, 0.00
+37, 8.50, -88.31, 0.00
```
Comment (pcarruscag) [inline TestCases/py_wrapper/translating_NACA0012/forces_0.csv.ref:39]:
```suggestion
37, 8.50, -88.30, 0.00
```

File: TestCases/py_wrapper/translating_NACA0012/forces_0.csv.ref [hunk 4]
```diff
@@ -1,200 +1,200 @@
...
+38, 9.24, -98.33, 0.00
+39, 9.95, -108.56, 0.00
```
Comment (pcarruscag) [inline TestCases/py_wrapper/translating_NACA0012/forces_0.csv.ref:41]:
```suggestion
39, 9.95, -108.57, 0.00
```

File: TestCases/py_wrapper/translating_NACA0012/forces_0.csv.ref [hunk 5]
```diff
@@ -1,200 +1,200 @@
...
+40, 10.62, -118.95, 0.00
+41, 11.24, -129.51, 0.00
+42, 11.81, -140.28, 0.00
+43, 12.33, -151.21, 0.00
+44, 12.78, -162.28, 0.00
+45, 13.15, -173.41, 0.00
```
Comment (pcarruscag) [inline TestCases/py_wrapper/translating_NACA0012/forces_0.csv.ref:47]:
```suggestion
45, 13.15, -173.40, 0.00
```

File: TestCases/py_wrapper/translating_NACA0012/forces_0.csv.ref [hunk 6]
```diff
@@ -1,200 +1,200 @@
...
+49, 13.72, -218.38, 0.00
+50, 13.58, -229.47, 0.00
+51, 13.30, -240.10, 0.00
+52, 12.85, -250.14, 0.00
+53, 12.31, -261.02, 0.00
+54, 11.62, -271.62, 0.00
+55, 10.32, -270.49, 0.00
+56, 8.26, -248.20, 0.00
+57, 6.94, -245.95, 0.00
+58, 7.97, -348.30, 0.00
```
Comment (pcarruscag) [inline TestCases/py_wrapper/translating_NACA0012/forces_0.csv.ref:60]:
```suggestion
57, 6.94, -245.94, 0.00
58, 7.97, -348.31, 0.00
```

File: TestCases/py_wrapper/translating_NACA0012/forces_0.csv.ref [hunk 7]
```diff
@@ -1,200 +1,200 @@
...
+59, 7.61, -440.57, 0.00
+60, 5.11, -448.31, 0.00
+61, 2.28, -436.77, 0.00
+62, -0.53, -428.04, 0.00
+63, -3.36, -418.60, 0.00
+64, -6.17, -407.62, 0.00
+65, -8.93, -395.29, 0.00
+66, -11.62, -381.67, 0.00
+67, -14.20, -367.12, 0.00
+68, -16.64, -351.57, 0.00
```
Comment (pcarruscag) [inline TestCases/py_wrapper/translating_NACA0012/forces_0.csv.ref:70]:
```suggestion
68, -16.64, -351.56, 0.00
```

File: TestCases/py_wrapper/translating_NACA0012/forces_0.csv.ref [hunk 8]
```diff
@@ -1,200 +1,200 @@
...
+149, 49.29, 784.46, 0.00
+150, 52.17, 786.78, 0.00
+151, 54.87, 788.13, 0.00
+152, 57.40, 788.54, 0.00
+153, 59.77, 788.13, 0.00
+154, 61.95, 786.74, 0.00
+155, 63.49, 778.73, 0.00
+156, 67.16, 797.62, 0.00
+157, 62.15, 716.23, 0.00
+158, 1.28, 14.31, 0.00
```
Comment (pcarruscag) [inline TestCases/py_wrapper/translating_NACA0012/forces_0.csv.ref:160]:
```suggestion
157, 62.16, 716.24, 0.00
158, 1.28, 14.32, 0.00
```

**Pull request merged**